libclc/generic/lib/math/cosh.cl

```diff
@@ -132,7 +132,7 @@ _CLC_OVERLOAD _CLC_DEF double cosh(double x) {
     // z = cosh(y) = cosh(y0)cosh(dy) + sinh(y0)sinh(dy)
     // where sinh(y0) and cosh(y0) are tabulated above.
 
-    int ind = min((int)y, 36);
+    int ind = min((int)(long)y, 36);
```

Comment:
is `nearbyint` available and any better semantically?

Reply:
on second thoughts: probably not since they still return a floating point value. The other alternative is `rint`, but I think it's allowed to raise an FPE, so what you have is probably best

Reply:
I don't believe it's available in OpenCL C as per the spec, strictly speaking. It may be that we can hack it in since we're in control of the compilation process, but if in doubt I think it's better to adhere to as close to OpenCL C as possible.

Reply:
There's also https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html#explicit-conversions `convert_int`, but again - this is probably fine

Reply:
I believe even the explicit conversions in OpenCL C have the same out-of-range behaviour: https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html#out-of-range-behavior

We'd have to do a `sat` conversion (`convert_int_sat`) but that'd be slower. I started out with that but noticed some other instances of the `(int)(long)` pattern in `libclc` so thought I'd be consistent.

Reply:
Right. Makes sense. I'm totally into this